src/segment-loader.js

```diff
@@ -74,6 +74,46 @@ export const illegalMediaSwitch = (loaderType, startingMedia, newSegmentMedia) =
 };
 
 /**
+ * Calculates a time value that is safe to remove from the back buffer without interupting
+ * playback.
+ *
+ * @param {TimeRange} seekable
+ *        The current seekable range
+ * @param {Number} currentTime
+ *        The current time of the player
+ * @param {Number} targetDuration
+ *        The target duration of the current playlist
+ * @return {Number}
+ *         Time that is safe to remove from the back buffer without interupting playback
+ */
+export const safeBackBufferTrimTime = (seekable, currentTime, targetDuration) => {
+  // Don't allow removing from the buffer within target duration of current time
+  // to avoid the possibility of removing the GOP currently being played which could
+  // cause playback stalls.
+  const safeRemoveToTimeLimit = currentTime - targetDuration;
```

Comment:
This line and comment might be better just as part of the `Math.min` return. We can remove the variable since it just obfuscates the meaning (since it is just a subtraction of two numbers).